wallet: accept language option for mnemonic when creating new wallet

pinheadmz opened the pull request:
Closes https://github.com/handshake-org/hsd/issues/587

Once https://github.com/handshake-org/hs-client/pull/39 is merged together:

```sh
$ hsw-cli --network=regtest --id=cartera --language=spanish mkwallet
{
  "network": "regtest",
  "wid": 1,
  "id": "cartera",
  "watchOnly": false,
  "accountDepth": 1,
  "token": "251e4f2b6a238cf3491ee783bc2d0ab09a8aeed8fb81296cc81d1bbd026c1812",
  "tokenDepth": 0,
  "master": {
    "encrypted": false
  },
  "balance": {
    "account": -1,
    "tx": 0,
    "coin": 0,
    "unconfirmed": 0,
    "confirmed": 0,
    "lockedUnconfirmed": 0,
    "lockedConfirmed": 0
  }
}

$ hsw-cli --network=regtest --id=cartera master
{
  "encrypted": false,
  "key": {
    "xprivkey": "rprvKE8qsHtkmUxUSR1stB2VJX3ibU3WzvUWZXxrzTJph7Sy91AVQJwBPVdE9HxZEkH6MfeDME26YLtmUZ46p1EDpn2HV16rpFGyJ81A6ikrKxV4"
  },
  "mnemonic": {
    "bits": 256,
    "language": "spanish",
    "entropy": "2ff615cbe0665107f3f28665fee8ab7755500159a6c22118b80f0bc94f99d62e",
    "phrase": "chancla perder inicio radical goma lote saber este grada vencer cara tribu faltar abono grave silencio borde reja acusar aula muela ronco fiebre sapo"
  }
}

```

nodech reviewed (commented):
So this only applies when you want wallet to generate mnemonic. (w/o passing phrase)
Otherwise if mnemonic has the phrase it will be used to detect language.

Check is becomes kinda weird: if mnemonic is null becomes the only case when `language` will be applied, so `if (!(mnemonic instanceof Mnemonic))` will only happen when mnemonic is `null/undefined`?

In that case it can be rewritten as `if (!mnemonic)`..

Unless manually called with `string` type as in test `should create wallet with spanish mnemonic` in `wallet-test`.

nodech commented on lib/wallet/wallet.js:
```diff
@@ -107,7 +108,12 @@ class Wallet extends EventEmitter {
         'Must create wallet with hd private key.');
     } else {
-      mnemonic = new Mnemonic(options.mnemonic);
+      if (!(mnemonic instanceof Mnemonic)) {
```
so to reiterate, this could be rewritten to be more precise to our use case (to be more true to what gets passed around)
and I think it makes this more readable.
```js
  if (typeof mnemonic === 'string')
    mnemonic = new Mnemonic({ phrase: mnemonic });

  // null or undefined
  if (mnemonic == null) // or if (!mnemonic) to include all falsy values?
    mnemonic = new Mnemonic({ language: options.language }); 
```

pinheadmz replied:
agreed, thanks - I cleaned this up

pinheadmz commented on test/wallet-test.js:
```diff
@@ -134,6 +135,24 @@ describe('Wallet', function() {
+    for (const word of phrase.split(' ')) {
+      const language = Mnemonic.getLanguage(word);
+      assert.strictEqual(language, 'spanish');
+      // Comprobar la cordura
```
Sanity check!

pinheadmz commented:
Rebased on master and also added an explicit error if an unknown language is requested:

(still requires https://github.com/handshake-org/hs-client/pull/39 to test)

```
--> hsw-cli mkwallet --id=testgerman --language='german'
Error: Unknown language.
    at WalletClient.request (/Users/matthewzipkin/Desktop/work/hs-client/node_modules/bcurl/lib/client.js:233:19)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async CLI.createWallet (/Users/matthewzipkin/Desktop/work/hs-client/bin/hsw-cli:87:20)
    at async CLI.handleWallet (/Users/matthewzipkin/Desktop/work/hs-client/bin/hsw-cli:552:9)
    at async /Users/matthewzipkin/Desktop/work/hs-client/bin/hsw-cli:676:3
--> hsw-cli mkwallet --id=testfrench --language='french'
{
  "network": "regtest",
  "wid": 9,
  "id": "testfrench",
  "watchOnly": false,
  "accountDepth": 1,
  "token": "16caa2a60a2d89bae1198b5dfc38641ab2a5d4a31b146294309f272850e64c1b",
  "tokenDepth": 0,
  "master": {
    "encrypted": false
  },
  "balance": {
    "account": -1,
    "tx": 0,
    "coin": 0,
    "unconfirmed": 0,
    "confirmed": 0,
    "lockedUnconfirmed": 0,
    "lockedConfirmed": 0
  }
}
--> hsw-cli --id=testfrench master
{
  "encrypted": false,
  "key": {
    "xprivkey": "rprvKE8qsHtkmUxUSQ1ukQb3FaszkrjHb1YNgDmDNoP8DxduM3rKC4pB2r7PaiH7k7uDSWArQj4BFm6WeD2Uzc46VdaXnss93awT1YtrzpTv7uTE"
  },
  "mnemonic": {
    "bits": 256,
    "language": "french",
    "entropy": "3bfd09f4877232dd0f218d1b524fd1303cc0c4d271ee80d6515a3aa94b051d56",
    "phrase": "cruel tasse hurler antidote bricoler furieux cuisine circuler bélier luisant usage exister remuer falaise doseur cylindre inductif paresse enviable gratuit maléfice inoculer gratuit somnoler"
  }
}
```
